### PR TITLE
[NSTK-1931] Ephemeral volume doc and examples

### PR DESCRIPTION
**Pull request by haibinxie**

### CODE REVIEW

File: docs/csi-ephemeral-volume.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# CSI Ephemeral Volume
+
+## Introduction
+Traditionally, CSI volumes are used with a PersistentVolume and PersistentVolumeClaim object combination, whose lifecycle is independent of Pods. Ephemeral volume feature allows CSI volumes to be specified directly in the pod specification.  The lifecycle of ephemeral volumes is bound to pod, they are created when pod is created and deleted when pod is terminated.
+
+## Check if Ephemeral Volume is enabled
```
Comment (dsupure) [inline docs/csi-ephemeral-volume.md:6]: [nit] Maybe change to "Check if Ephemeral Volume feature is enabled"?

File: docs/csi-ephemeral-volume.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+After installing PSO, MODES should contain Ephemeral. 
+```bash
+/k8s# kubectl get csidriver
+NAME       ATTACHREQUIRED   PODINFOONMOUNT   MODES                  AGE
+pure-csi   true             true             Persistent,Ephemeral   28h
+```
+
+## Ephemeral Volume Naming Convention
+Ephemeral volume does not have corresponding objects (like PV/PVC) in Kubernetes. Ephemeral volume name is prefixed with PSO cluster ID, pod name and pod namespace. In this format:
```
Comment (dsupure) [inline docs/csi-ephemeral-volume.md:15]: [nit/suggestion] 
"Ephemeral volume does not have corresponding objects (like PV/PVC) in Kubernetes. The ephemeral volume name is prefixed with PSO clusterID, pod name, and pod namespace using the following format:"

File: docs/examples/ephemeralvolume/pod-ephemeral-volume-file.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      driver: pure-csi
+      volumeAttributes:
+        backend: file
+        # Optional export rules
+        # exportrules: "*(rw)"
+        size: 2Gi
+  containers:
+  - name: nginx
+    image: nginx
+    # Configure a mount for the ephemeral volume
```
Comment (dsupure) [inline docs/examples/ephemeralvolume/pod-ephemeral-volume-file.yaml:19]: [nit/suggtion] # Configure a mount point for the ephemeral volume

File: docs/examples/ephemeralvolume/pod-ephemeral-volume-block.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      driver: pure-csi
+      # File system types: vfat, xfs, ext4
+      fsType: xfs
+      volumeAttributes:
+        backend: block
+        size: 2Gi
+  containers:
+  - name: nginx
+    image: nginx
+    # Configure a mount for the ephemeral volume
```
Comment (dsupure) [inline docs/examples/ephemeralvolume/pod-ephemeral-volume-block.yaml:19]: [nit/suggtion] # Configure a mount point for the ephemeral volume

**Comment by haibinxie** [comment]
@dsupure  comments resolved.

**Review by dsupure — APPROVED** [review]
LGTM!

**Comment by dsupure** [comment]
@sdodsley : Can you help to take a second pass? Thanks.

File: docs/csi-ephemeral-volume.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# CSI Ephemeral Volume
+
+## Introduction
+Traditionally, CSI volumes are used with a PersistentVolume and PersistentVolumeClaim object combination, whose lifecycle is independent of Pods. Ephemeral volume feature allows CSI volumes to be specified directly in the pod specification.  The lifecycle of ephemeral volumes is bound to pod, they are created when pod is created and deleted when pod is terminated.
```
Comment (sdodsley) [inline docs/csi-ephemeral-volume.md:4]: The Ephemersal Volume feature...
...bound to the pod, being created...

File: docs/csi-ephemeral-volume.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+## Check if Ephemeral Volume feature is enabled
+After installing PSO, MODES should contain Ephemeral. 
+```bash
+/k8s# kubectl get csidriver
+NAME       ATTACHREQUIRED   PODINFOONMOUNT   MODES                  AGE
+pure-csi   true             true             Persistent,Ephemeral   28h
+```
+
+## Ephemeral Volume Naming Convention
+Ephemeral volume does not have corresponding objects (like PV/PVC) in Kubernetes. The ephemeral volume name is prefixed with PSO clusterID, pod name and pod namespace with the following format:
```
Comment (sdodsley) [inline docs/csi-ephemeral-volume.md:15]: An Ephemeral volume...

File: docs/csi-ephemeral-volume.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+```
+(pso-cluster-id)-(pod-name)-(pod-namespace)-(unique-suffix)
+```
+
+## Examples:
+**Note:** Ephemeral volume does not support raw block.
```
Comment (sdodsley) [inline docs/csi-ephemeral-volume.md:21]: volumes do not support

**Review by sdodsley — CHANGES_REQUESTED** [review]

**Comment by haibinxie** [comment]
@sdodsley comments resolved.

File: docs/csi-ephemeral-volume.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# CSI Ephemeral Volume
+
+## Introduction
+Traditionally, CSI volumes are used with a PersistentVolume and PersistentVolumeClaim object combination, whose lifecycle is independent of Pods. The ephemeral volume feature allows CSI volumes to be specified directly in the pod specification.  The lifecycle of ephemeral volumes is bound to the pod, being created when pod is created and deleted when pod is terminated.
```
Comment (sdodsley) [inline docs/csi-ephemeral-volume.md:4]: being created when the pod ...
and deleted when the pod ...

**Review by sdodsley — CHANGES_REQUESTED** [review]
Last nits

File: docs/csi-ephemeral-volume.md (hunk 8 above)
Comment (haibinxie) [reply docs/csi-ephemeral-volume.md]: @sdodsley resolved too

**Review by sdodsley — APPROVED** [review]

**Pull request merged**